game/main.py

```diff
@@ -214,6 +214,20 @@ def getOpenSpaces(self) -> List[Masu]:
                     masuList.append(masu)
         return masuList
 
+    def kingIsInCheck(self, isSente: bool) -> bool:
+        kingCoords = self.findKingCoordinates(isSente)
+        if kingCoords is None:
+            return False
+        opposingPieces = list(filter(lambda x: x[0].isSente != isSente, self.getPieces()))
+        for opposingPiece in opposingPieces:
+            piece = opposingPiece[0]
+            masu = opposingPiece[1]
+            moves = piece.legalMoves(self, masu)

```

Comment:
there should be some consideration here for the fact that pieces that are pinned can still deny squares from the enemy king